docs/CustomRoutes.md

```diff
@@ -57,34 +115,105 @@ const Settings = () => (
+    <Link component={RouterLink} to="/settings">
+        Settings
+    </Link>
+);
+```
+
+Alternately, create a [custom menu](./Menu.md) with entries for the custom pages.
+
+```jsx
+// in src/MyMenu.js
+import { Menu } from 'react-admin';
+import SettingsIcon from '@mui/icons-material/Settings';
+import PeopleIcon from '@mui/icons-material/People';
+
+export const MyMenu = () => (
+    <Menu>
+        <Menu.DashboardItem />
+        <Menu.ResourceItem to="/posts"  />
+        <Menu.ResourceItem to="/comments" />
+        <Menu.Item to="/settings" primaryText="Users" leftIcon={<SettingsIcon />}/>
+        <Menu.Item to="/profile" primaryText="Miscellaneous" leftIcon={<PeopleIcon />}/>
+    </Menu>
+);
+```
+
+## Sub-Routes
+
+If you want to add sub-routes to a resource, add the `<Route>` elements as [children of the `<Resource>` element](./Resource.md#children):
+
+```jsx
+import { Admin, Resource } from 'react-admin';
 import { Route } from "react-router-dom";
-import Settings from './Settings';
-import Register from './register';
+
+import dataProvider from './dataProvider';
+import posts from './posts';
 
 const App = () => (
-    <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
-        <CustomRoutes noLayout>
-            <Route path="/register" element={<Register />} />
-        </CustomRoutes>
-        <CustomRoutes noLayout>
-            <Route path="/settings" element={<Settings />} />
-        </CustomRoutes>
+    <Admin dataProvider={dataProvider}>
+        <Resource name="posts" {...posts}>
+            <Route path="analytics" element={<PostAnalytics/>} />
+        </Resource>
     </Admin>
 );
 
-export default App;
+// is equivalent to
+const App = () => (
+    <Admin dataProvider={dataProvider}>
+        <Resource name="posts" {...posts} />
+        <CustomRoutes>
+            <Route path="/posts/analytics" element={<PostAnalytics />} />
+        </CustomRoutes>
+    </Admin>
+);
 ```
 
-When a user browses to `/register`, the `<Register>` component will appear outside the defined Layout, leaving you the freedom to design the screen the way you want.
+This is usually useful for nested resources, such as books on authors:
+
+{% raw %}
+```jsx
+// in src/App.jss
+import { Admin, Resource, ListGuesser, EditGuesser } from 'react-admin';
+import { Route } from "react-router-dom";
 
-**Tip**: Custom routes can be [a `<Navigate>` route](https://reactrouter.com/docs/en/v6/api#navigate), too.
+const App = () => (
+    <Admin dataProvider={dataProvider}>
+        <Resource name="authors" list={ListGuesser} edit={EditGuesser}>
+            <Route path=":authorId/books" element={<BookList />} />
+        </Resource>
+    </Admin>
+);
+
+// in src/BookList.jss
+import { useParams } from 'react-router-dom';
+import { List, Datagrid, TextField } from 'react-admin';
+
+const BookList = () => {
+    const { authorId } = useParams();
+    return (
+        <List resource="books" filter={{ authorId }}>
```

Comment:
A comment here could be nice to explain why we need to specify the `resource` prop